src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -610,15 +617,16 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
                                                     updatedMethodSymbolMapOpt,
                                                     snapshotBuilderOpt);
 
-            Analyze(walker, symbol, diagnostics, initialState, snapshotBuilderOpt);
+            Analyze(walker, symbol, diagnostics, initialState, snapshotBuilderOpt, out variableTypes);
         }
 
         private static void Analyze(
             NullableWalker walker,
             Symbol symbol,
             DiagnosticBag diagnostics,
             VariableState initialState,
-            SnapshotManager.Builder snapshotBuilderOpt)
+            SnapshotManager.Builder snapshotBuilderOpt,
+            out ImmutableDictionary<Symbol, TypeWithAnnotations> variableTypes)
```

Comment:
>out ImmutableDictionary<Symbol, TypeWithAnnotations> variableTypes [](start = 12, length = 66)

Consider having the caller pass in an optional mutable `Dictionary<Symbol, TypeWithAnnotations>` so we're not allocating for callers that ignore this value.